lxd/networks_utils.go

```diff
@@ -659,6 +660,63 @@ func networkFanAddress(underlay *net.IPNet, overlay *net.IPNet) (string, string,
 	return fmt.Sprintf("%s/%d", ipBytes.String(), overlaySize), dev, ipStr, err
 }
 
+func networkKillForkDNS(name string) error {
+	// Check if we have a running dnsmasq at all
+	pidPath := shared.VarPath("networks", name, "forkdns.pid")
+	if !shared.PathExists(pidPath) {
+		return nil
+	}
+
+	// Grab the PID
+	content, err := ioutil.ReadFile(pidPath)
+	if err != nil {
+		return err
+	}
+	pid := strings.TrimSpace(string(content))
+
+	// Check for empty string
+	if pid == "" {
+		os.Remove(pidPath)
+		return nil
+	}
+
+	// Check if the process still exists
+	if !shared.PathExists(fmt.Sprintf("/proc/%s", pid)) {
+		os.Remove(pidPath)
+		return nil
+	}
+
+	// Check if it's dnsmasq
+	cmdArgs, err := ioutil.ReadFile(fmt.Sprintf("/proc/%s/cmdline", pid))
```

Comment:
Yeah, I think you should drop the check before and only perform this check. :)

Reply:
done